### PR TITLE
Support fixture in repository-gcs:thirdPartyTest and fix GCS fixture

### PR DESCRIPTION
**Pull request by andrershov**
It turns out that today `:plugins:repository-gcs:thirdPartyTest` can only run against real GCS.
Moreover, thirdPartyTest is not a part of `check`, so these tests are not running on intake build.
This commit addresses the issue and makes `repository-gcs:thirdPartyTest` work with both fixture and real GCS.
To do that, except adjusting build and test itself, I had to make changes to the fixture, because previously it was ignoring `BlobListOption.currentDirectory()` in the list call.
This is a ground needed to add GCS fixture based testing to the snapshot-tool.

### CODE REVIEW

**Comment by original-brownbear** [comment]
@andrershov FYI, builds failing here.

**Comment by andrershov** [comment]
@original-brownbear test failure should be fixed now

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2 (watcher)

**Review by original-brownbear — CHANGES_REQUESTED** [review]
Looks ok in general, a few smaller comments.

Did you test this against real GCS to make sure it still works?

File: plugins/repository-gcs/build.gradle [hunk 1]
```diff
@@ -125,21 +125,55 @@ check {
   dependsOn 'qa:google-cloud-storage:check'
 }
 
+test {
+  exclude '**/GoogleCloudStorageThirdPartyTests.class'
+}
+
+task thirdPartyTest(type: Test) {
```
[the remark this answers is not in the document]
Comment (andrershov) [reply plugins/repository-gcs/build.gradle]: Sure, https://github.com/elastic/elasticsearch/pull/44885/commits/39fdfb324f32b9502e9d54d83cf24e86d0ce3df1

File: plugins/repository-gcs/build.gradle [hunk 2]
```diff
@@ -125,21 +125,55 @@ check {
...
+    nonInputProperties.systemProperty 'test.google.account', "${ -> encodedCredentials.call() }"
+    nonInputProperties.systemProperty 'test.google.endpoint', "${ -> fixtureEndpoint.call() }"
+    nonInputProperties.systemProperty 'test.google.tokenURI', "${ -> tokenURI.call() }"
+  }
+} else if (!gcsServiceAccount || !gcsBucket || !gcsBasePath) {
+  throw new IllegalArgumentException("not all options specified to run thirdPartyTest against external GCS service are present")
+} else {
+  thirdPartyTest {
+    include '**/GoogleCloudStorageThirdPartyTests.class'
+    systemProperty 'tests.security.manager', 'false'
```
[the remark this answers is not in the document]
Comment (andrershov) [reply plugins/repository-gcs/build.gradle]: Agreed, it works fine with security manager enabled https://github.com/elastic/elasticsearch/pull/44885/commits/39433d8ede3ba4da879cb255fc6ba640a6be4140

File: plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageThirdPartyTests.java [hunk 3]
```diff
@@ -40,6 +41,24 @@
         return pluginList(GoogleCloudStoragePlugin.class);
     }
 
+    @Override
+    protected Settings nodeSettings() {
+        Settings.Builder builder = Settings.builder();
```
[the remark this answers is not in the document]
Comment (andrershov) [reply plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageThirdPartyTests.java]: https://github.com/elastic/elasticsearch/pull/44885/commits/417a2f690e37881e380a717a693dd1f5c6e3bf85

**Comment by andrershov** [comment]
@original-brownbear thanks for your review! I've made the necessary changes. Of course, I've made sure that it works for both real GCS and fixture. Ready for the next pass.

**Review by original-brownbear — APPROVED** [review]
thanks @andrershov LGTM if we get CI to be happy :)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Pull request merged**